File: Leetcode 3701-3800/3713/3713.py
# Leetcode 3713: Longest Balanced Substring I
# https://leetcode.com/problems/longest-balanced-substring-i/
# Solved on 27th of December, 2025
class Solution:
    def longestBalanced(self, s: str) -> int:
        """
        Finds the length of the longest balanced substring.

        Args:
            s: The input string.
        Returns:
            The length of the longest balanced substring.
        """
        n = len(s)

        maxLen = 0
        for i in range(n):
            freq = {}

            for j in range(i, n):
                char = s[j]
                freq[char] = freq.get(char, 0) + 1
                counts = list(freq.values())

                if len(set(counts)) == 1:
                    maxLen = max(maxLen, j - i + 1)

        return maxLen

File: Leetcode 3701-3800/3713/3713.java
// Leetcode 3713: Longest Balanced Substring I
// https://leetcode.com/problems/longest-balanced-substring-i/
// Solved on 27th of December, 2025
class Solution {
    /**
     * Finds the length of the longest balanced substring.
     * A substring is balanced if all characters in it appear the same number of times.
     * @param s The input string.
     * @return The length of the longest balanced substring.
     */
    public int longestBalanced(String s) {
        int n = s.length();
        int maxLength = 0;
        for (int i = 0; i < n; i++) {
            int[] freq = new int[26];
            int maxFreq = 0;
            int distinctCount = 0;
            for (int j = i; j < n; j++) {
                int index = s.charAt(j) - 'a';
                if (freq[index] == 0) {
                    distinctCount++;
                }
                freq[index]++;
                maxFreq = Math.max(maxFreq, freq[index]);
                
                if ((j - i + 1) == (maxFreq * distinctCount)) {
                    maxLength = Math.max(maxLength, j - i + 1);
                }
            }
        }
        return maxLength;
    }
}